documentation/specs/proposed/General_perf_onepager.md

```diff
@@ -0,0 +1,42 @@
+# General performance improvements
+MSBuild is the main tool used for building .NET projects. It is used by Visual Studio, the .NET CLI, and other build tools. The performance of MSBuild is critical to the productivity of developers. This document outlines our goals to improve overall performance of MSBuild.
+
+## Goals and Motivation
+
+We are aiming for:
+ - Searching opportunities to improve the build times.
+ - Reducing MSBuild overhead in the Visual Studio IDE.
+ - Collecting data to understand the current performance bottlenecks.
+
+This document does not cover specific performance improvements, but rather outlines the general goals and motivation for the performance improvements.
+
+
+## Impact
+    
+    - Opening project/solution, branch switching and other operations using MSBuild code in VS should be less impacted by the MSBuild.
+    - Overall build times should be reduced. 
+    - Even a small improvement can save a lot of time and computing resources across all builds done daily worldwide.
+
+## Stakeholders
+
+    - Chet Hust (PM) - as a customer advocate
+    - David Kean - as a VS performance expert
+
+## Risks
+
+    - Performance improvements might not be as significant as expected.
+    - We can break existing functionality while making changes.
+    - We can thow away some ideas due to technical limitations or complexity/improvements ratio.
```

Comment:
```suggestion
    - Some ideas and performance measurement findings might need to be skipped due to technical limitations or complexity/improvements ratio.
```

Reply:
One way to limit this is timebox; start with the real easy wins, if you are spending too much time on one win, just move onto the next one until you run out. I'm super confident based on working in this code base, looking at traces and perf in general, we will find more easy wins than we'll be able to fix.